cloud-sql-proxy/pyspark_metastore_test.py

```diff
@@ -24,25 +34,54 @@
 
 sc = pyspark.SparkContext()
 sqlContext = pyspark.sql.HiveContext(sc)
-
-# Find available table name.
-table_names = sqlContext.tableNames()
-test_table_name = None
-while not test_table_name or test_table_name in table_names:
-  test_table_name = 'table_' + ''.join(
+def configure_table_mysql():
+  # Find available table name.
+  table_names = sqlContext.tableNames()
+  test_table_name = None
+  while not (test_table_name or test_table_name in table_names):
+    test_table_name = 'table_' + ''.join(
       [random.choice(string.ascii_lowercase) for x in range(4)])
 
-# Create table.
-sqlContext.range(10).write.saveAsTable(test_table_name)
+  # Create table.
+  sqlContext.range(10).write.saveAsTable(test_table_name)
 
-# Read table metadata from Cloud SQL proxy.
-tables = sqlContext.read.jdbc(
+  # Read table metadata from Cloud SQL proxy.
+  tables = sqlContext.read.jdbc(
     'jdbc:mysql:///{}?user={}&password={}'.format(METASTORE_DB, HIVE_USER,
                                                   HIVE_USER_PASSWORD), 'TBLS')
-test_table = tables.where(tables.TBL_NAME == test_table_name).collect()[0]
+  test_table = tables.where(tables.TBL_NAME == test_table_name).collect()[0]
 
-print('Successfully found table {} in Cloud SQL Hive metastore'.format(
+  print('Successfully found table {} in Cloud SQL Hive metastore'.format(
     test_table.TBL_NAME))
 
-# Clean up table.
-sqlContext.sql('DROP TABLE ' + test_table_name)
+  # Clean up table.
+  sqlContext.sql('DROP TABLE ' + test_table_name)
+
+
+def configure_table_postgresql():
+  # Find available table name.
+  table_names = sqlContext.tableNames()
+  test_table_name = None
+  while not (test_table_name or test_table_name in table_names):
+    test_table_name = 'table_' + ''.join(
+      [random.choice(string.ascii_lowercase) for x in range(4)])
+
+  # Create table.
+  sqlContext.range(10).write.saveAsTable(test_table_name)
+
+  # Read table metadata from Cloud SQL proxy.
+  tables = sqlContext.read.jdbc(
+    'jdbc:postgresql:///{}?user={}&password={}'.format(METASTORE_DB, HIVE_USER,
+                                                  HIVE_USER_PASSWORD), 'TBLS')
+  test_table = tables.where(tables.TBL_NAME == test_table_name).collect()[0]
+
+  print('Successfully found table {} in Cloud SQL Hive metastore'.format(
+    test_table.TBL_NAME))
+
+  # Clean up table.
+  sqlContext.sql('DROP TABLE ' + test_table_name)
+
+if args.instance_type == INSTANCE_MYSQL:
+  configure_table_mysql()
+elif args.instance_type == INSTANCE_POSTGRESQL:
+  configure_table_postgresql()
```

Comment:
the test logic is the same, can we deduplicated the code here
something like extract the jdbc connection string out, something like 
jdbcUri = if mysql jdbc:mysql://.... else jdbc:postgres...
then test with the jdbcUri?